packages/vitest/src/node/watch-filter.ts

```diff
@@ -0,0 +1,160 @@
+    stdout().write(`${c.cyan('?')} ${c.bold(this.message)} › `)
+    return new Promise((resolve) => {
+      this.onKeyPress = this.filterHandler(filterFunc, (result) => {
+        this.close()
+        resolve(result)
+      })
+      process.stdin.on('keypress', this.onKeyPress)
+    })
+  }
+
+  private filterHandler(filterFunc: FilterFunc, onSubmit: (result?: string) => void) {
+    return async (str: string, key: any) => {
+      switch (true) {
+        case key.sequence === '\x7F':
+          if (this.currentKeyword && this.currentKeyword?.length > 1)
+            this.currentKeyword = this.currentKeyword?.slice(0, -1)
+
+          else
+            this.currentKeyword = undefined
+
+          break
+        case key?.ctrl && key?.name === 'c':
+        case key?.name === 'escape':
+          this.cancel()
+          onSubmit(undefined)
+          break
+        case key?.name === 'enter':
+        case key?.name === 'return':
+          onSubmit(this.results[this.selectionIndex] || this.currentKeyword || '')
+          this.currentKeyword = undefined
+          break
+        case key?.name === 'up':
+          if (this.selectionIndex && this.selectionIndex > 0)
+            this.selectionIndex--
+          else
+            this.selectionIndex = -1
+
+          break
+        case key?.name === 'down':
+          if (this.selectionIndex < this.results.length - 1)
+            this.selectionIndex++
+          else if (this.selectionIndex >= this.results.length - 1)
+            this.selectionIndex = this.results.length - 1
+
+          break
+        case !key?.ctrl && !key?.meta:
+          if (this.currentKeyword === undefined)
+            this.currentKeyword = str
+
+          else
+            this.currentKeyword += str
+          break
+      }
+
+      if (this.currentKeyword)
+        this.results = await filterFunc(this.currentKeyword)
+
+      this.render()
+    }
+  }
+
+  private render() {
+    this.printKeyword()
+    if (!this.currentKeyword) {
+      this.eraseAndPrint('\nPlease input filter pattern')
+    }
+    else if (this.currentKeyword && this.results.length === 0) {
+      this.eraseAndPrint(`\nPattern matches no results`)
+    }
+    else {
+      const resulCountLine = `Pattern matches ${this.results.length} results`
+
+      let resultBody = ''
+
+      if (this.results.length > MAX_RESULT_COUNT) {
+        const offset = this.selectionIndex > SELECTION_MAX_INDEX ? this.selectionIndex - SELECTION_MAX_INDEX : 0
+        const displayResults = this.results.slice(offset, MAX_RESULT_COUNT + offset)
+        const remainingResultCount = this.results.length - offset - displayResults.length
+
+        resultBody = `${displayResults.map((result, index) => (index + offset === this.selectionIndex) ? c.green(` › ${result}`) : c.dim(` › ${result}`)).join('\n')}`
+        if (remainingResultCount > 0)
+          resultBody += '\n' + `${c.dim(`   ...and ${remainingResultCount} more results`)}`
```

Comment:
if there is only 1 more, it shows "1 more result**s**" instead of "1 more result"

Reply:
Thanks!
I fixed it. (6a0355d)